### PR TITLE
Add localization

### LINKED ISSUE #63: Annotating a parameter with  `@Name` does not reflect on Discord.
**Describe the bug**
Using the `Name` decorator does not reflect on Discord end and use the parameter name as argument.

**To Reproduce**
```dart
final testCommand = ChatCommand(
  'hi',
  'A test command',
  (IChatContext ctx, @Name('awawa') String foo) =>
     ctx.respond(MessageBuilder.content('A'),
  ),
);

// Command registration and all that stuff..
```
Currently, this is rendered as `/hi foo`

**Expected behavior**
And should be rendered as `/hi awawa`

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Linux
 - Dart version Stable: 2.17.1
 - Nyxx version 4..0.0-dev.2
 - nyxx_commands version 4.2.0

**Additional context**
I think this is caused by this line when using `dart:mirrors`:
https://github.com/nyxx-discord/nyxx_commands/blob/73b6015b075834339537aa2b60e7e0fc18050bb8/lib/src/mirror_utils/with_mirrors.dart#L39

### PR DESCRIPTION
**Pull request by Rapougnac**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
Add localization support as it is implemented in `_interactions`

Also fixes #63 

### Remark
I had to use my fork as a dependency as application overwrites are in `main` and locales are in `dev`.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze .`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: bin/compile/function_metadata/metadata_builder.dart
Comment (abitofevrything) [inline bin/compile/function_metadata/metadata_builder.dart:126]: If no localized names are passed, this will result in the normal name being selected.
I.e `@Name('foo')` will result in both `name` and `localizedNames` being set to `foo`.

File: bin/compile/function_metadata/metadata_builder.dart
Comment (abitofevrything) [inline bin/compile/function_metadata/metadata_builder.dart:136]: Same issue as with the localized names.

File: lib/src/mirror_utils/with_mirrors.dart
Comment (abitofevrything) [inline lib/src/mirror_utils/with_mirrors.dart:48]: Should have a check to ensure there is at most one `@Name` annotation like the other annotations.

**Review by abitofevrything — CHANGES_REQUESTED** [review]

File: bin/compile/function_metadata/metadata_builder.dart
Comment (Rapougnac) [reply bin/compile/function_metadata/metadata_builder.dart:126]: What's the point of doing this? If it's null, first argument will be taken as fallback.
That would be redundant too i think.

Comment (abitofevrything) [reply bin/compile/function_metadata/metadata_builder.dart]: It's an issue because if `localizedNames` ends up being the non-localized name expression, the generated file will attempt to put a `String` expression into an argument of type `Map<Locale, String>` in the parameter data.

Comment (Rapougnac) [reply bin/compile/function_metadata/metadata_builder.dart]: Ohh, okay, my bad, I misread your sentence, thought that you were asking to add this behaviour.

File: lib/src/commands/chat_command.dart
Comment (abitofevrything) [inline lib/src/commands/chat_command.dart:464]: Maybe we should check localized name validity here too - also applies to User and Message Commands

File: lib/src/commands/options.dart
Comment (abitofevrything) [inline lib/src/commands/options.dart:16]: Unnecessary import

File: lib/src/commands/chat_command.dart
Comment (abitofevrything) [reply lib/src/commands/chat_command.dart:464]: Ah yes, just for chat commands then.

**Review by abitofevrything — COMMENTED** [review]
Looks good, will merge when nyxx_interactions releases and this PR starts to use the new version.

File: pubspec.yaml
Comment (abitofevrything) [inline pubspec.yaml:33]: Once this is removed we're ready to merge!

**Review by abitofevrything — CHANGES_REQUESTED** [review]

**Review by abitofevrything — APPROVED** [review]

**Pull request merged**
